### DISCUSSION TITLE
function uuid_generate_v4() does not exist (SQLSTATE 42883)

### DISCUSSION THREAD

**Discussion by igorbrasileiro** [supabase/supabase#10512] [Questions] [answered]
Hi, i'm using https://github.com/supabase/supa_audit to audit some tables in my supabase project.

My auth.users table has this trigger below to insert new users into public.profiles table. My login flow let create users `shouldCreateUser: true`, so when the user login first time, it creates a user that triggers insert a profile, then triggers to insert the audit.

```sql
create or replace function public.handle_new_user()
returns trigger
language plpgsql
security definer set search_path = public
as $$
begin
  insert into public.profiles (user_id, name, email)
  values (new.id, new.email, new.email);
  return new;
end;
$$;

create trigger on_auth_user_created
  after insert on auth.users
  for each row execute procedure public.handle_new_user();
```

the public profile table has triggers from supa_audit, that uses this uuid_generate_v4(); Supa audit table and functions are in the schema "audit";

```
failed to close prepared statement: ERROR: current transaction is aborted, commands ignored until end of transaction block (SQLSTATE 25P02): ERROR: function uuid_generate_v4() does not exist (SQLSTATE 42883)
```

When I open the supabase SQL editor and run `select uuid_generate_v4()` it works.

Versions:
supabase@1.35.4
supa_audit@0.3.0

**Comment by igorbrasileiro** [comment, marked as answer]
The trigger has the security definer set search_path = public, i removed this and everything worked well.
> ```sql
> create or replace function public.handle_new_user()
> returns trigger
> language plpgsql
> security definer set search_path = public
> as $$
> begin
>   insert into public.profiles (user_id, name, email)
>   values (new.id, new.email, new.email);
>   return new;
> end;
> $$;

I'm little bit newbe, after read the postgresql documentation I still don't understanding.

**Reply by GaryAustin1** [reply]
You need extensions in the search path also.

**Reply by igorbrasileiro** [reply]
I didn't added supa_audit via extensions, i copied the supa_audit-0.3.0.sql file and ran. So, i need to add the `search_path = audit`?

**Reply by GaryAustin1** [reply]
generate_uuid_v4() is in the pgcrypto extension which is in extensions.

**Reply by igorbrasileiro** [reply]
```sql
create or replace function public.handle_new_user()
returns trigger
language plpgsql
security definer set search_path = public, extensions
as $$
begin
  insert into public.profiles (user_id, name, email)
  values (new.id, new.email, new.email);
  return new;
end;
$$;
```

worked well, thank you.
